### DISCUSSION TITLE
Testing template

### DISCUSSION THREAD

**Discussion by tpluscode** [copier-org/copier#580] [Q&A]
How does one test a template? A way to provide various combinations of answers and checks of the generated files

**Comment by pawamoy** [comment]
I don't have something similar to a test matrix, but maybe you can checkout my tests in https://github.com/pawamoy/copier-pdm and build on top of it. Would be quite interested in the result.

**Comment by yajo** [comment]
Copier has an extensive pytest suite. You can borrow inspiration from there too.
